src/gui_common/SpeciesDetailsPanel.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using Godot;
+
+public class SpeciesDetailsPanel : PanelContainer
+{
+    [Export]
+    public NodePath SpeciesDetailsLabelPath = null!;
+
+    [Export]
+    public NodePath SpeciesPreviewPath = null!;
+
+    [Export]
+    public NodePath HexPreviewPath = null!;
+
+    [Export]
+    public NodePath FossilisationButtonPath = null!;
+
+    [Export]
+    public NodePath FossilisationDialogPath = null!;
+
+    private CustomRichTextLabel speciesDetailsLabel = null!;
+    private SpeciesPreview speciesPreview = null!;
+    private CellHexesPreview hexesPreview = null!;
+    private Button fossilisationButton = null!;
+    private FossilisationDialog fossilisationDialog = null!;
+
+    private Species? previewSpecies;
+
+    public Species? PreviewSpecies
+    {
+        get => previewSpecies;
+        set
+        {
+            previewSpecies = value;
```

Comment:
This should be rewritten to work when set even before this is attached to the scene tree.

Reply:
I've used a `ready` bool but I seem to remember you requesting a different approach to this in the past. I can't remember what that was. Let me know if you had a better approach in mind.

Reply:
Make the control nullable, then check in the update method if it is null, and if it is then skip. And finally also add a call to that update method at the end of _Ready